### PR TITLE
Added CSS styling and fixed reconnecting to backend without reload.

### PR DESCRIPTION
**Pull request by chris-dietz**
- Added CSS to match existing web client look.
- Changed timeout on websockets to 1 second (1000 ms)
- Also made the client reset and disable input when the websocket connection goes down.

Should be the final bit that needs to be done to replace the old web client.

### CODE REVIEW

**Review by ia0 — COMMENTED** [review]
Thanks! It looks pretty good to final merge.

The only issue I have is that we shouldn't clear the logs when disconnecting (just print that we disconnected), and only clear when connecting. In particular I'd like the `applet rust panic` example to actually show the applet message and the panic message. I tried to remove the `history.clear()` but this doesn't seem to be enough. There's maybe some timing issue. When I add `timer::sleep_ms(2000)` before the `panic!("ready to die")` in `examples/rust/panic/src/lib.rs` then it works (the history is present after disconnect).

One way to fix this would be for the host runner to stay alive, but it could die for other reasons, so it would be nice if the log is not cleared when disconnected.

**Comment by ia0** [comment]
By the way, I've fixed an issue with `xtask` not using the correct applet. You should merge `dev/web-ui` in your branch before testing switching between `hello`, `panic`, and `button_abort` examples.

**Comment by ia0** [comment]
And once this PR is merged, you can create a PR from dev/web-ui to main with https://github.com/google/wasefire/compare/dev/web-ui?expand=1.

File: crates/runner-host/crates/web-client/src/app.rs
Comment (chris-dietz) [reply crates/runner-host/crates/web-client/src/app.rs]: Fixed

File: crates/runner-host/crates/web-client/src/console.rs
Comment (chris-dietz) [reply crates/runner-host/crates/web-client/src/console.rs]: Makes sense I removed clearing on disconnect.

File: crates/runner-host/crates/web-client/src/hooks.rs
Comment (chris-dietz) [reply crates/runner-host/crates/web-client/src/hooks.rs:52]: We don't, good point! Fixed

**Comment by chris-dietz** [comment]
> Thanks! It looks pretty good to final merge.
> 
> The only issue I have is that we shouldn't clear the logs when disconnecting (just print that we disconnected), and only clear when connecting. In particular I'd like the `applet rust panic` example to actually show the applet message and the panic message. I tried to remove the `history.clear()` but this doesn't seem to be enough. There's maybe some timing issue. When I add `timer::sleep_ms(2000)` before the `panic!("ready to die")` in `examples/rust/panic/src/lib.rs` then it works (the history is present after disconnect).
> 
> One way to fix this would be for the host runner to stay alive, but it could die for other reasons, so it would be nice if the log is not cleared when disconnected.

Huh, I'm having trouble replicating that on my end. It seems to keep the history when I run the panic applet now. I was having a issue where the server would relaunch the web app even though it was already open sometimes. I fixed it by increasing the time the server waits for the client to connect. (Changing the retry interval on the client didn't have an effect, I think there's just too much overhead for it to reconnect within 2 seconds consistently.)

Maybe that fixed the problem you were having as well? Could you try again and see if it's still happening?

**Comment by chris-dietz** [comment]
Also I made trunk build the web client in "release" mode now. Hopefully that will make things faster and possibly fix some timing issues.

File: crates/runner-host/crates/web-client/src/console.rs
Comment (ia0) [inline crates/runner-host/crates/web-client/src/console.rs:85]: nit

```suggestion
                <input disabled={!*button_enabled}  type="submit" value="Send" />
```

File: crates/runner-host/crates/web-client/src/hooks.rs
Comment (ia0) [inline crates/runner-host/crates/web-client/src/hooks.rs:45]: nit
```suggestion
    match &*command_state {
        None | Some(Command::Disconnected) => (),
        _ => command_state.set(Some(Command::Disconnected)),
    }
```

File: crates/runner-host/crates/web-client/data/index.scss
Comment (ia0) [inline crates/runner-host/crates/web-client/data/index.scss:25]: Is it possible to add some margin (or any other CSS technique) to get the buttons and LEDs horizontally centered in those 79 pixels of the left panel?

**Review by ia0 — COMMENTED** [review]
LGTM thanks!

Regarding the 2 to 3 seconds change, can you try again with 2 seconds? On my side I didn't get any issue. And I would prefer to reduce the starting time (which affects all users all the time) than increase it to fix possible reopens (which affects less users some of the time).

If trunk doesn't build in release by default, it's definitely needed to do that. And speaking of release, maybe there's too much logging to the console. Instead of use `wasm_logger::Config::default()` we should use `wasm_logger::Config::new(level)` with appropriate level. This probably needs to be done at compile-time using features (one per `log::Level`) and set in the host runner when building trunk using `log::max_level().to_level()` to know which feature to use. If no feature is passed (i.e. when `to_level()` returns `None`), then logging should be completely disabled. But this should be done in a separate PR.

File: crates/runner-host/crates/web-client/data/index.scss
Comment (chris-dietz) [reply crates/runner-host/crates/web-client/data/index.scss:30]: Definitely, should be centered now.

**Comment by chris-dietz** [comment]
> LGTM thanks!
> 
> Regarding the 2 to 3 seconds change, can you try again with 2 seconds? On my side I didn't get any issue. And I would prefer to reduce the starting time (which affects all users all the time) than increase it to fix possible reopens (which affects less users some of the time).
> 
> If trunk doesn't build in release by default, it's definitely needed to do that. And speaking of release, maybe there's too much logging to the console. Instead of use `wasm_logger::Config::default()` we should use `wasm_logger::Config::new(level)` with appropriate level. This probably needs to be done at compile-time using features (one per `log::Level`) and set in the host runner when building trunk using `log::max_level().to_level()` to know which feature to use. If no feature is passed (i.e. when `to_level()` returns `None`), then logging should be completely disabled. But this should be done in a separate PR.

I tried it again and it doesn't seem to be an issue anymore. So I reverted to 2 seconds. 

And yeah I agree, I'll set the logging level based on compilation flags. I'll create a follow up PR doing that. :)

**Review by ia0 — APPROVED** [review]
Thanks!

Once merged, do you mind [creating a PR](https://github.com/google/wasefire/compare/main...dev/web-ui?expand=1) from `dev/web-ui` to `main`? This way you will be the main author. We can also decide to make you code owner of the `dev/web-ui` if that's a role you're willing to take.

The PR for logging can still be based on `dev/web-ui` and updated to `main` once the `dev/web-ui` is merged in `main`.

**Comment by chris-dietz** [comment]
> Thanks!
> 
> Once merged, do you mind [creating a PR](https://github.com/google/wasefire/compare/main...dev/web-ui?expand=1) from `dev/web-ui` to `main`? This way you will be the main author. We can also decide to make you code owner of the `dev/web-ui` if that's a role you're willing to take.
> 
> The PR for logging can still be based on `dev/web-ui` and updated to `main` once the `dev/web-ui` is merged in `main`.

PR is is created. :) 

And yeah I would have no problems with taking ownership over the 'dev/web-ui' branch. ^^

**Pull request merged**
